compactor panic on startup with memberlist

Following from #3057, compactor also panics on startup if configured to use memberlist.
```
level=info ts=2020-08-25T11:32:50.771154347Z caller=main.go:162 msg="Starting Cortex" version="(version=1.3.0, branch=master, revision=5ea4cc6aa)"
level=info ts=2020-08-25T11:32:50.771487504Z caller=server.go:223 http=[::]:3100 grpc=[::]:9095 msg="server listening on addresses"
level=debug ts=2020-08-25T11:32:50.771690226Z caller=api.go:94 msg="api: registering route" methods= path=/config auth=false
level=debug ts=2020-08-25T11:32:50.771743321Z caller=api.go:94 msg="api: registering route" methods= path=/ auth=false
level=debug ts=2020-08-25T11:32:50.771865118Z caller=api.go:94 msg="api: registering route" methods= path=/compactor/ring auth=false
level=debug ts=2020-08-25T11:32:50.771964952Z caller=api.go:94 msg="api: registering route" methods= path=/services auth=false
level=info ts=2020-08-25T11:32:50.772124805Z caller=module_service.go:58 msg=initialising module=compactor
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x29a81d0]

goroutine 312 [running]:
github.com/cortexproject/cortex/pkg/ring/kv.createClient(0xc00021ff00, 0xa, 0x3746a39, 0xb, 0x374f191, 0xe, 0x0, 0x0, 0x4a817c800, 0x0, ...)
        /go/src/github.com/cortexproject/cortex/pkg/ring/kv/client.go:145 +0x6a0
github.com/cortexproject/cortex/pkg/ring/kv.NewClient(0xc00021ff00, 0xa, 0x3746a39, 0xb, 0x374f191, 0xe, 0x0, 0x0, 0x4a817c800, 0x0, ...)
        /go/src/github.com/cortexproject/cortex/pkg/ring/kv/client.go:122 +0x132
github.com/cortexproject/cortex/pkg/ring.NewLifecycler(0xc00021ff00, 0xa, 0x3746a39, 0xb, 0x374f191, 0xe, 0x0, 0x0, 0x4a817c800, 0x0, ...)
        /go/src/github.com/cortexproject/cortex/pkg/ring/lifecycler.go:152 +0x201
github.com/cortexproject/cortex/pkg/compactor.(*Compactor).starting(0xc00022e600, 0x3f2cd80, 0xc0008dccc0, 0x1, 0xc00043f300)
        /go/src/github.com/cortexproject/cortex/pkg/compactor/compactor.go:203 +0x4c3
github.com/cortexproject/cortex/pkg/util/services.(*BasicService).main(0xc0009ec5a0)
        /go/src/github.com/cortexproject/cortex/pkg/util/services/basic_service.go:134 +0x459
created by github.com/cortexproject/cortex/pkg/util/services.(*BasicService).StartAsync.func1
        /go/src/github.com/cortexproject/cortex/pkg/util/services/basic_service.go:96 +0xa8
```

Tested with image `master-5ea4cc6aa`.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -28,7 +28,7 @@
 * [BUGFIX] Querier: Merge results from chunks and blocks ingesters when using streaming of results. #3013
 * [BUGFIX] Querier: query /series from ingesters regardless the `-querier.query-ingesters-within` setting. #3035
 * [BUGFIX] Experimental blocks storage: Ingester is less likely to hit gRPC message size limit when streaming data to queriers. #3015
-* [BUGFIX] Experimental blocks storage: fixed memberlist support for the store-gateways ring used when blocks sharding is enabled. #3058
+* [BUGFIX] Experimental blocks storage: fixed memberlist support for the store-gateways and compactors ring used when blocks sharding is enabled. #3058 #3095
 * [BUGFIX] Fix configuration for TLS server validation, TLS skip verify was hardcoded to true for all TLS configurations and prevented validation of server certificates. #3030
 * [BUGFIX] Fixes the Alertmanager panicking when no `-alertmanager.web.external-url` is provided. #3017
 * [BUGFIX] Fixes the registration of the Alertmanager API metrics `cortex_alertmanager_alerts_received_total` and `cortex_alertmanager_alerts_invalid_total`. #3065

--- a/pkg/cortex/modules.go
+++ b/pkg/cortex/modules.go
@@ -664,7 +664,7 @@ func (t *Cortex) setupModuleManager() error {
 		Ruler:          {Overrides, Distributor, Store, StoreQueryable, RulerStorage},
 		Configs:        {API},
 		AlertManager:   {API},
-		Compactor:      {API},
+		Compactor:      {API, MemberlistKV},
 		StoreGateway:   {API, Overrides, MemberlistKV},
 		Purger:         {Store, DeleteRequestsStore, API},
 		All:            {QueryFrontend, Querier, Ingester, Distributor, TableManager, Purger, StoreGateway, Ruler},

